lua/lazyvim/plugins/extras/lang/r.lua

```diff
@@ -7,64 +7,53 @@ return {
-                f = { name = "+(f)unctions", ["🚫"] = "which_key_ignore" },
-                g = { name = "+(g)oto", ["🚫"] = "which_key_ignore" },
-                k = { name = "+(k)nit", ["🚫"] = "which_key_ignore" },
-                p = { name = "+(p)aragraph", ["🚫"] = "which_key_ignore" },
-                r = { name = "+(r) general", ["🚫"] = "which_key_ignore" },
-                s = { name = "+(s)plit or (s)end", ["🚫"] = "which_key_ignore" },
-                t = { name = "+(t)erminal", ["🚫"] = "which_key_ignore" },
-                v = { name = "+(v)iew", ["🚫"] = "which_key_ignore" },
-              },
-            })
-          end,
-        },
-        min_editor_width = 72,
-        rconsole_width = 78,
-        disable_cmds = {
-          "RClearConsole",
-          "RCustomStart",
-          "RSPlot",
-          "RSaveClose",
-        },
-        synctex = false,
-        pdfviewer = "open",
-      }
-      -- Check if the environment variable "R_AUTO_START" exists.
-      -- If using fish shell, you could put in your config.fish:
-      -- alias r "R_AUTO_START=true nvim"
-      if vim.env.R_AUTO_START == "true" then
-        opts.auto_start = "on startup"
-        opts.objbr_auto_start = true
-      end
-      require("r").setup(opts)
-    end,
-    lazy = false,
+          -- Increase the width of which-key to handle the longer r-nvim descriptions
+          local wk = require("which-key")
+          wk.setup({
+            layout = {
+              width = { min = 20, max = 100 }, -- min and max width of the columns
+            },
+          })
+          -- Workaround from https://github.com/folke/which-key.nvim/issues/514#issuecomment-1987286901
+          wk.register({
+            ["<localleader>"] = {
+              a = { name = "+(a)ll", ["🚫"] = "which_key_ignore" },
+              b = { name = "+(b)etween marks", ["🚫"] = "which_key_ignore" },
+              c = { name = "+(c)hunks", ["🚫"] = "which_key_ignore" },
+              f = { name = "+(f)unctions", ["🚫"] = "which_key_ignore" },
+              g = { name = "+(g)oto", ["🚫"] = "which_key_ignore" },
+              k = { name = "+(k)nit", ["🚫"] = "which_key_ignore" },
+              p = { name = "+(p)aragraph", ["🚫"] = "which_key_ignore" },
+              r = { name = "+(r) general", ["🚫"] = "which_key_ignore" },
+              s = { name = "+(s)plit or (s)end", ["🚫"] = "which_key_ignore" },
+              t = { name = "+(t)erminal", ["🚫"] = "which_key_ignore" },
+              v = { name = "+(v)iew", ["🚫"] = "which_key_ignore" },
+            },
+          })
+        end,
+      },
+      pdfviewer = (function()
+        if vim.loop.os_uname().sysname:find("Windows", 1, true) then
+          return "open" -- according to @Aman9das on Github
+        elseif vim.loop.os_uname().sysname:find("Darwin", 1, true) then
+          return "open"
+        else
+          return "xdg-open" -- according to @Aman9das on Github
+        end
+      end)(),
```

Comment:
Instead of all these, you could use `vim.ui.open` I believe. It returns the default open handler based on the OS. See `:h vim.ui.open()` for more info.

Reply:
That's a great idea, but sadly the `pdfviewer` parameter only takes a path to an application (or something defined in `$PATH`). Since `vim.ui.open()` is new in vim 0.10, I wonder if there are any plans on incorporating it into `R.nvim` directly.

I also checked out `vim.ui.open()` and didn't see any direct way to get the application it uses to open, which would be really nice for this scenario.

<img width="609" alt="Screenshot 2024-05-23 at 4 11 14 PM" src="https://github.com/LazyVim/LazyVim/assets/54869170/36d6b0bd-0132-49c6-954d-e44fabe3407b">

Reply:
Not entirely sure myself to be honest, but could you try
```lua
      pdfviewer = (function()
        return vim.ui.open
      end)(),
```
Otherwise, maybe the maintainer will have a better idea if this is possible.

Reply:
That was a great idea, but sadly not 😞 

<img width="605" alt="Screenshot 2024-05-24 at 11 39 25 PM" src="https://github.com/LazyVim/LazyVim/assets/54869170/13f131b8-4c64-4135-b136-783875d8f183">

Reply:
I don't know then, sorry. Waiting for maintainer's input would the best :)

Reply:
Could you also try `pdfviewer = vim.ui.open` or `pdfviewer = vim.ui.open()`? Just throwing out ideas :stuck_out_tongue_closed_eyes:

Reply:
Should we ship it as is with the longer, manually specified per-OS code, or hopefully wait for their implementation? I've made a [PR](https://github.com/R-nvim/R.nvim/pull/144) for `R.nvim` to bring up this idea with a simple implementation that hopefully I can work with some people to properly implement it.

I am fine shipping it now and then creating a new PR to update LazyVim then when/if I implement `vim.ui.open` in the `R.nvim`.

Reply:
You just have to wait for the maintainer's approval. I'm just a regular user like you.

Reply:
I think shipping this asis is good. Then working on issues filed against the extra will be more effective way of improving it.